modules/apps/headless/headless-admin-content/headless-admin-content-impl/src/main/java/com/liferay/headless/admin/content/internal/resource/v1_0/StructuredContentResourceImpl.java

```diff
@@ -28,4 +47,67 @@
 )
 public class StructuredContentResourceImpl
 	extends BaseStructuredContentResourceImpl {
+
+	@Override
+	public Page<StructuredContent> getSiteStructuredContentsVersionInformationPage(
+		@NotNull Long siteId, Boolean flatten, String search,
+		Aggregation aggregation, Filter filter, Pagination pagination,
+		Sort[] sorts) throws Exception {
+		return SearchUtil.search(null,
+			booleanQuery -> {
+			},
+			filter, JournalArticle.class, search, pagination,
+			queryConfig -> queryConfig.setSelectedFieldNames(
+				Field.ARTICLE_ID,
+				Field.SCOPE_GROUP_ID),
+			searchContext -> {
+				searchContext.addVulcanAggregation(aggregation);
+				searchContext.setAttribute(
+					com.liferay.portal.kernel.search.Field.STATUS,
+					WorkflowConstants.STATUS_APPROVED);
+				searchContext.setAttribute("head", Boolean.TRUE);
+				searchContext.setCompanyId(contextCompany.getCompanyId());
+
+				if (siteId != null) {
+					searchContext.setGroupIds(new long[]{siteId});
+				}
+
+			}, sorts,
+			document -> {
+				JournalArticle journalArticle =
```

Comment:
Let's extract this to a method to be cleaner

Reply:
AFAIK, I think Brian prefers to avoid use methods only once. But we can extract this to a method because it is pretty sure that it will necessary in the future

Reply:
In this case we have this pattern in all ResourceImpl when using the SearchUtil we always use a internal method called _to{DTOName} to make the transformation. 

I think it is cleaner this way and Brian won't say anything in this case. Brian's rule of thump I think it is that if the name is not clear he prefers to inline it. But if it is a good name there is no problem

Reply:
Ok, I will extract to a method then. Thanks 😊